src/onefetch/cli_utils.rs

```diff
@@ -0,0 +1,94 @@
+use crate::onefetch::{ascii_art::AsciiArt, error::*, info::Info, language::Language};
+use colored::Color;
+use std::env;
+use std::io::Write;
+use strum::IntoEnumIterator;
+
+pub struct Printer<W> {
+    writer: W,
+    info: Info,
+}
+
+impl<W: Write> Printer<W> {
+    pub fn new(writer: W, info: Info) -> Self {
+        Self { writer, info }
+    }
+
+    pub fn print(&mut self) -> Result<()> {
+        let center_pad = "   ";
+        let info_str = format!("{}", &self.info);
+        let mut info_lines = info_str.lines();
+        let colors: Vec<Color> = Vec::new();
+        let mut buf = String::new();
+        if let Some(custom_image) = &self.info.config.image {
+            if let Some(image_backend) = &self.info.config.image_backend {
+                buf.push_str(&image_backend.add_image(
+                    info_lines.map(|s| format!("{}{}", center_pad, s)).collect(),
+                    custom_image,
+                ));
+            } else {
+                panic!("No image backend found")
+            }
+        } else {
+            let mut logo_lines = if let Some(custom_ascii) = &self.info.config.ascii_input {
+                AsciiArt::new(custom_ascii, &colors, !self.info.config.no_bold)
+            } else {
+                AsciiArt::new(
+                    self.get_ascii(),
+                    &self.info.colors,
+                    !self.info.config.no_bold,
+                )
+            };
+
+            loop {
+                match (logo_lines.next(), info_lines.next()) {
+                    (Some(logo_line), Some(info_line)) => {
+                        buf.push_str(&format!("{}{}{:^}\n", logo_line, center_pad, info_line))
+                    }
+                    (Some(logo_line), None) => buf.push_str(&format!("{}\n", logo_line)),
+                    (None, Some(info_line)) => buf.push_str(&format!(
+                        "{:<width$}{}{:^}\n",
+                        "",
+                        center_pad,
+                        info_line,
+                        width = logo_lines.width()
+                    )),
+                    (None, None) => {
+                        buf.push_str("\n");
```

Comment:
This might result in a clippy error.
```suggestion
                        buf.push('\n');
```